### ISSUE TITLE
Lightning Metric return NaN

### ISSUE THREAD

**Issue by XDynames** [XDynames/modular-encoder-decoders-segmentation#3]
Current dropped in the pytorch-lightning metric.classification.IoU module which returns NaN for both training and validation after ~10 epochs.

Haven't looked into how ignore label / ignore background works yet but this would be my first point of interrogation
Other than that it might have something to do with how they handle the numerical softening to prevent division by zero

**Comment by AdrianOrenstein** [comment]
The way that i've implemented ignore label in the past was masking in the values I want i.e.
ignore_mask = (labels == -100)
result = f(pred[~ignore_mask], target[~ignore_mask])

We'll have to open their stuff up and see what's going wrong
